File: tests/fixtures/go_sample/infrastructure/user_repo.go
package infrastructure

import (
	"database/sql"
	"github.com/example/gosample/domain"
)

// UserRepo is the infrastructure implementation for user storage.
type UserRepo struct {
	db *sql.DB
}

// FindUser retrieves a user from the database.
func (r *UserRepo) FindUser(id int) *domain.User {
	return &domain.User{ID: id, Name: "Bob"}
}

File: tests/fixtures/java_sample/src/infrastructure/UserRepo.java
package com.example.javasample.infrastructure;

import com.example.javasample.domain.User;
import java.util.List;

/**
 * UserRepo is the infrastructure implementation for user storage.
 */
public class UserRepo {
    public User findUser(int id) {
        return new User(id, "Bob");
    }

    public List<User> findAll() {
        return List.of(new User(1, "Alice"), new User(2, "Bob"));
    }
}
